charmcraft/commands/build.py

```diff
@@ -175,8 +177,12 @@ def handle_dependencies(self):
 
     def handle_package(self):
         """Handle the final package creation."""
+        logger.debug("Parsing the project's metadata")
+        with (self.charmdir / CHARM_METADATA).open('rt', encoding='utf8') as fh:
+            metadata = yaml.safe_load(fh)
+
         logger.debug("Creating the package itself")
-        zipname = self.charmdir.name + '.charm'
+        zipname = metadata['name'] + '.charm'
```

Comment:
I definitely agree that the charm name should come from the metadata name and not the directory. That is the name that Juju will deploy it as.